packages/overlays/src/OverlayMixin.js

```diff
@@ -15,52 +16,110 @@ export const OverlayMixin = dedupeMixin(
             type: Boolean,
             reflect: true,
           },
-          popperConfig: Object,
+          config: {
+            type: Object,
+          },
+          closeEventName: {
+            type: String,
+          },
         };
       }
 
+      constructor() {
+        super();
+        this.config = {};
+        this.closeEventName = 'overlay-close';
```

Comment:
what is the use case for separate close event names? how about reusing the OverlayController events? e.g. show/hide/before-show/before-hide?